src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -524,3 +533,77 @@ def update_os_patch_configuration_sub_setting(self, patch_configuration_sub_sett
     def do_processes_require_restart(self):
         """Defaulting this for Apt"""
         return False
+
+    def is_reboot_pending(self):
+        """ Checks if there is a pending reboot on the machine. """
+        try:
+            pending_file_exists = os.path.isfile(self.REBOOT_PENDING_FILE_PATH)
+            pending_processes_exists = self.do_processes_require_restart()
+            self.composite_logger.log_debug(" - Reboot required debug flags: " + str(pending_file_exists) + ", " + str(
+                pending_processes_exists) + ".")
+            is_reboot_required = pending_file_exists or pending_processes_exists
+
+            # Use Ubuntu Pro Client api in case of Ubuntu VMs.
+            if not self.__pro_client_prereq_met:
+                self.composite_logger.log_debug("Ubuntu Pro Client pre-requisite not met.")
+                return is_reboot_required
+
+            is_pro_api_success, pro_reboot_status = self.ubuntu_pro_client.is_reboot_pending()
+            self.composite_logger.log_debug(
+                "is_pro_api_success: {0}, reboot_status {1}".format(is_pro_api_success, pro_reboot_status))
+
+            # Ubuntu Pro Client api failed. return default status.
+            if not is_pro_api_success:
+                return is_reboot_required
+
+            # Compare Ubuntu Pro Client result and default result. Log if there is mismatch.
+            if is_reboot_required != pro_reboot_status:
+                self.composite_logger.log_debug(
+                    "Reboot status mismatch. pro = {0}, default = {1}".format(pro_reboot_status,
+                                                                              is_reboot_required))
+
+            # Ubuntu Pro Client result is considered truth even if there is mismatch.
+            return pro_reboot_status
+        except Exception as error:
+            self.composite_logger.log_error('Error while checking for reboot pending: ' + repr(error))
+            return True  # defaults for safety
+
+    def __is_pro_client_prereq_met(self):
+        """check required conditions to use Ubuntu Pro Client"""
+        if Constants.UbuntuProClientSettings.FEATURE_ENABLED \
+                and self.__get_os_major_version() <= Constants.UbuntuProClientSettings.MAX_OS_MAJOR_VERSION_SUPPORTED \
+                and self.__is_minimum_required_python_installed() \
+                and self.ubuntu_pro_client.is_pro_working():
+            return True
+        else:
+            # Log each condition for debug.
+            self.composite_logger.log_debug("Feature_Enabled = {0}, OS Version = {1}, python {2}".format(Constants.UbuntuProClientSettings.FEATURE_ENABLED, self.__get_os_major_version() <= Constants.UbuntuProClientSettings.MAX_OS_MAJOR_VERSION_SUPPORTED, self.__is_minimum_required_python_installed()))
```

Comment:
self.__get_os_major_version() <= Constants.UbuntuProClientSettings.MAX_OS_MAJOR_VERSION_SUPPORTED

This will log a boolean instead of the OS Version

Reply:
Noticed the same in python log too. If the intention is to log whether the conditions are met, the text part could be:
"ProClientCompatibility checks: IsFeatureEnabled = {0}, IsOSVersionCompatible = {1}, IsPythonCompatible {2}"

Reply:
yes, the log for os version and python version is already available elsewhere. The intention is to find which condition failed